lib/codeowners/checker.rb

```diff
@@ -93,13 +112,36 @@ def defined_owner?(file)
       false
     end
 
+    def valid_owner?(owner)
+      return true unless @validate_owners
+
+      owners_list.include?(owner)
+    end
+
     def codeowners
       @codeowners ||= CodeOwners.new(
         FileAsArray.new(codeowners_filename),
         transformers: transformers
       )
     end
 
+    def owners_list
+      return [] unless @validate_owners
+
+      @owners_list ||=
+        if github_credentials_exist?
```

Comment:
Can we change the order to load the file first and if it's missing - use Github?

@jonatas what do you think?

Reply:
That is good for the CI to use this order but for local fast checkers without an internet connection, it would be fast. I agree with both ways.